tensorflow/lite/kernels/mul.cc

```diff
@@ -183,6 +184,22 @@ TfLiteStatus EvalQuantized(TfLiteContext* context, TfLiteNode* node,
           TF_LITE_MUL(optimized_integer_ops, Mul, int8_t);
         }
       }
+    } else if (input1->type == kTfLiteInt16) {
+      // We have this check, because in case of int16
+      // input1_val*input2_val can overflow int32:
+      // see MulElementwise -
+      // tensorflow/lite/kernels/internal/reference/integer_ops/mul.h in case of
+      // 16-bit this function is used in symmetric quantization, so offset
+      // should be zero.
+      TF_LITE_ENSURE_EQ(context, op_params.input1_offset, 0.0);
```

Comment:
are you sure symmetric quantization is the only case?

didn't see the specifications: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/tools/optimize/operator_property.cc#L772-L775

Reply:
Hi @renjie-liu, 
Thank you for the review.

We introduce this kernel for the quantization scheme: activations in 16-bit and weights in 8-bit. The range of 16-bit is quite large and it is sufficient, if we do only symmetric quantization. That's why we consider only symmetric quantization for 16-bit reference kernels.
Should operator specifications reflect this case ?  Should we have a special "restricted_value" property for 16-bit that shows that zero point is zero in this case ?

Reply:
For 16bit we are only supporting symmetric for now.